src/media/js/site/containers/tos.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {signTOS} from '../actions/tos';
+
+
+export class TOSStub extends React.Component {
+  render() {
+    const styles = {
+      background: '#FFF',
+      padding: '1em 2em',
+      margin: '0 -60px',
+    };
+    return (
+      <section id="tos" style={styles}>
+        <h1>FPO Terms of Service</h1>
+        <p>
+          When run in one of our regular environments, the standalone TOS will
+          be displayed here, and it will be prettier than this. For now, some
+          Lorem Ipsum:
+        </p>
+        <p>
+          Sed facilisis enim ac magna suscipit imperdiet. Integer tempor
+          tincidunt libero et gravida. Nulla et mollis libero. Fusce suscipit
+          lorem a leo venenatis, id auctor arcu consequat. Sed vitae volutpat
+          ex. Pellentesque habitant morbi tristique senectus et netus et
+          malesuada fames ac turpis egestas. Mauris velit neque, lobortis non
+          iaculis ut, euismod mollis tortor. Praesent interdum at mi at
+          tristique. Vestibulum et dui eget erat accumsan sodales nec et sem.
+          Phasellus id erat quis dui volutpat posuere vitae sit amet magna.
+          Nunc velit diam, tincidunt eu pretium ac, ullamcorper a nisi. Etiam
+          nisl nibh, laoreet eu vestibulum in, elementum ut elit. Maecenas
+          varius faucibus massa eu ultrices.
+        </p>
+      </section>
+    );
+  }
+}
+
+
+export default class TOSSignatureHandler extends React.Component {
+  static contextTypes = {
+    store: React.PropTypes.object
+  };
+
+  static propTypes = {
+    user: React.PropTypes.object,
+    signTOS: React.PropTypes.func.isRequired,
+  };
+
+  handleClick(evt) {
+    this.props.signTOS(this.dispatch, this.context.store.getState)
```

Comment:
you don't pass in dispatch and store manually. it won't do anything.

it's already binded with bindActionCreators, and it's handled by thunkMiddleware. so we can get rid of pulling in context